internal/utils.go

```diff
@@ -47,10 +48,38 @@ func getManifests(policyConf *types.PolicyConfig) ([]map[string]interface{}, err
 				manifestPaths = append(manifestPaths, yamlPath)
 			}
 		} else {
-			manifestPaths = append(manifestPaths, manifest.Path)
+			// Unmarshal the manifest in order to check for metadata patch replacement
+			manifestFile, err := unmarshalManifestFile(manifest.Path)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(*manifestFile) == 0 {
+				continue
+			}
+			// Allowing replace the original manifest metadata.name and/or metadata.namespace if it is a single
+			// yaml structure in the manifest path
+			if len(*manifestFile) == 1 && len(manifest.Patches) == 1 {
+				patchMetadata, ok := manifest.Patches[0]["metadata"].(map[string]interface{})
+				if ok {
+					metadata, ok := (*manifestFile)[0]["metadata"].(map[string]interface{})
+					if ok {
```

Comment:
Optional: This logic gets pretty nested in a bunch of if statements. How about something like this to make it a little easier to read?
```suggestion
				patchMetadata, patchOk := manifest.Patches[0]["metadata"].(map[string]interface{})
				metadata, ok := (*manifestFile)[0]["metadata"].(map[string]interface{})
				if patchOk && ok {
```